test/test-plugin/package.json

```diff
@@ -0,0 +1,19 @@
+{
+  "private": true,
+  "name": "test-plugin",
+  "version": "1.0.0",
+  "engines": {
+    "code-server": "^3.6.0"
```

Comment:
This annoyingly warns on every npm command, perhaps we should use a custom key?
```
warning code-server-example-plugin@0.0.1: The engine "code-server" appears to be invalid.
```

Reply:
So this is how vscode does things as well. I'm leaning in favour of keeping it the same to keep with the idiom but also ok changing to our own key.

cc @code-asher

Reply:
Cool, no worries if it's already convention

Reply:
It does seem to be a convention in use, as mentioned VS Code does it this way and I could have sworn I've seen it elsewhere although the details escape me. `--ignore-engines` can ignore it; I wonder if that's a valid setting for a `.yarnrc` or something.